src/DeploymentUtils.ts

```diff
@@ -1,19 +1,47 @@
-import * as deployments from "../deployments/deployments.json";
+import * as deployments_ from "../deployments/deployments.json";
+interface DeploymentExport {
+  [chainId: string]: { [contractName: string]: { address: string; blockNumber: number } };
+}
+const deployments: DeploymentExport = deployments_ as any;
 
 // Returns the deployed address of any contract on any network.
 export function getDeployedAddress(contractName: string, networkId: number): string {
   try {
-    return (deployments as any)[networkId.toString()][contractName].address;
+    return deployments[networkId.toString()][contractName].address; // First try match directly on the name provided.
   } catch (_) {
+    try {
+      // If this does not match then try search on a Regex on SpokePool to let the caller exclude the chain name.
+      for (const _contractName of Object.keys(deployments[networkId.toString()]))
+        if (/.*_SpokePool/.test(_contractName)) return deployments[networkId.toString()][_contractName].address;
+    } catch (_) {}
     throw new Error(`Contract ${contractName} not found on ${networkId} in deployments.json`);
   }
 }
 
 // Returns the deployment block number of any contract on any network.
-export function getDeployedBlockNumber(contractName: string, networkId: number): string {
+export function getDeployedBlockNumber(contractName: string, networkId: number): number {
   try {
-    return (deployments as any)[networkId.toString()][contractName].blockNumber;
+    return deployments[networkId.toString()][contractName].blockNumber;
   } catch (_) {
     throw new Error(`Contract ${contractName} not found on ${networkId} in deployments.json`);
   }
 }
+
+// Returns the chainId and contract name for a given contract address.
+export function getContractInfoFromAddress(contractAddress: string): { chainId: Number; contractName: string } {
```

Comment:
it looks like this solves my issues since im usually given the contract address and chain ( at least for spoke pools).   It would be a problem if hubpools also had different names, but they dont currently.

Reply:
ye, this method assumes that the deployment addresses on each chain are unique enabling you to go backwards from a known address to the contract name and chainId.

Reply:
Does the PolygonTokenBridger break this? It has the same address on multiple chains

Reply:
Ye that contract would not work with this method but that's ok. This method is only used in the bots to enhance logs and the bots never directly call those polygon contracts so it won't be an issue. 

I'll add an error that is thrown if multiple addresses are found to prevent any confusion